src/main/java/teammates/ui/webapi/SendLoginEmailAction.java

```diff
@@ -0,0 +1,60 @@
+package teammates.ui.webapi;
+
+import static teammates.common.util.FieldValidator.REGEX_EMAIL;
+
+import teammates.common.util.Const;
+import teammates.common.util.EmailSendingStatus;
+import teammates.common.util.EmailWrapper;
+import teammates.common.util.StringHelper;
+import teammates.ui.output.SendLoginEmailResponseData;
+import teammates.ui.request.InvalidHttpRequestBodyException;
+
+/**
+ * Sends a login email.
+ */
+class SendLoginEmailAction extends Action {
+
+    @Override
+    AuthType getMinAuthLevel() {
+        return AuthType.PUBLIC;
+    }
+
+    @Override
+    void checkSpecificAccessControl() throws UnauthorizedAccessException {
+        // Sending of login emails can be requested by anyone
+    }
+
+    @Override
+    public JsonResult execute() throws InvalidHttpRequestBodyException, InvalidOperationException {
+        String userEmail = getNonNullRequestParamValue(Const.ParamsNames.USER_EMAIL);
+        if (!StringHelper.isMatching(userEmail, REGEX_EMAIL)) {
+            throw new InvalidHttpParameterException("Invalid email address: " + userEmail);
+        }
+
+        String userCaptchaResponse = getRequestParamValue(Const.ParamsNames.USER_CAPTCHA_RESPONSE);
+        if (!recaptchaVerifier.isVerificationSuccessful(userCaptchaResponse)) {
+            return new JsonResult(new SendLoginEmailResponseData(false, "Something went wrong with "
+                    + "the reCAPTCHA verification. Please try again."));
+        }
+
+        String continueUrl = getNonNullRequestParamValue(Const.ParamsNames.CONTINUE_URL);
```

Comment:
Is this really non-nullable?

Reply:
I think so, because there is always a next URL, e.g., student home page, to go to upon successful login, unless there is a flow where users can just login without navigating to any page, e.g., remain in the main front page. I believe there is not one right?

Reply:
That's a front-end assumption; back-end API should not designed with that kind of assumption. Remember that back-end in theory is usable even without (web) front-end.

Reply:
Hm but does it make sense to be usable without front end when authentication happens in front end?

Reply:
I made a mistake in the elaboration previously. To clarify, `continueUrl` here refers to URL to be redirected to upon clicking the login link to complete the authentication process. This is different from `nextUrl`, i.e., the intended page that an user would like to access. Since Firebase Authentication is a frontend authentication solution, which always involve frontend logic, this param should be non-nullable.

Reply:
Indeed, `continueUrl` should be nullable.

But first, I made a mistake in my elaboration previously and would like to clarify that `continueUrl` is the URL to be redirected to upon clicking the login link to [complete the authentication process](https://firebase.google.com/docs/auth/web/email-link-auth#completing_sign-in_in_a_web_page). This is different from `nextUrl`, i.e., the intended protected page, which is what was misunderstood.

Since a `continueUrl` is a requirement by Firebase Authentication, I have abstracted out the arguments required to generate a login link via the `LoginLinkOptions` object. With the builder pattern, a `LoginLinkOptions` instance can be constructed depending on the `AuthService` in used.